### DISCUSSION TITLE
No signal

### DISCUSSION THREAD

**Discussion by hbarnard** [john30/ebusd#789] [Q&A]
Hi John

I expect this to be something simple/beginner. But. Followed all the Pi (it's a Pi2, latest 32 bit OS) installation and mount instructions (I haven't touched the jumper since arrival). 

Here's dmesg:

```
dmesg|grep ttyAMA0
[    3.238314] 3f201000.serial: ttyAMA0 at MMIO 0x3f201000 (irq = 114, base_baud = 0) is a PL011 rev2
```

Here's ebusd:
```
sudo ebusd --scanconfig -f  -d ens:/dev/ttyAMA0 --latency=50
2023-01-03 09:04:50.551 [main notice] ebusd 22.4.p20230102 started with auto scan on enhanced device /dev/ttyAMA0
2023-01-03 09:04:50.804 [bus notice] device status: resetting
2023-01-03 09:04:50.805 [bus notice] bus started with own address 31/36
2023-01-03 09:07:00.976 [main notice] update check: 
revision v22.4 available
```

and here's ebusctl:
```
ebusctl info
version: ebusd 22.4.p20230102
update check: 
revision v22.4 available
device: /dev/ttyAMA0, enhanced
signal: no signal
reconnects: 0
masters: 1
messages: 11
conditional: 0
poll: 0
update: 4
address 31: master #8, ebusd
address 36: slave #8, ebusd
```

I'm pretty sure the physical connection from the (Valliant) boiler works because I had an ESERA interface on it previously.  I've made sure that the interface is well seated on the Pi too.

What should I try next?

**Comment by DanielBuehner** [comment]
Same here. New RPi adapter but no signal. 

`pi@raspberrypi:~ $ dmesg|grep ttyAMA0
[    2.925762] 3f201000.serial: ttyAMA0 at MMIO 0x3f201000 (irq = 114, base_baud = 0) is a PL011 rev2`

`
pi@raspberrypi:~ $ ebusctl info
version: ebusd 22.4.v22.4
device: /dev/ttyAMA0, enhanced
signal: no signal
reconnects: 0
masters: 1
messages: 11
conditional: 0
poll: 0
update: 4
address 31: master #8, ebusd
address 36: slave #8, ebusd
`


Wondering if i have to put jumpers, but do i have to order them?!

**Comment by DanielBuehner** [comment]
I got the signal aquired. Please check if the serial0 is set to ttyAMA0. Within my first attemp the serial1 was set to ttyAMA0. After a few tries i got the serial0 to ttyAMA0. Then i got the signal aquired.

ls -l /dev 

and then searching for the mapping of serial0. This is usually either ttyAMA0 or ttyS0.

Since the ttyS0 UART has difficulties with higher baudrates, it is strongly recommended to definitely use ttyAMA0 which is using UART type PL011 where these problems cannot occur.

If serial0 is connected to ttyS0, which occurs mainly with the Raspberry Pi 3 and some Raspbian/Raspberry Pi OS versions, then ttyS0 and ttyAMA0 can be swapped by adding an entry in /boot/config.txt:

dtoverlay=pi3-miniuart-bt

followed by a reboot.

**Reply by hbarnard** [reply]
Thanks, unhappily (for me), this bit is correct in my set up:

`lrwxrwxrwx 1 root root           7 Jan  3 11:35 serial0 -> ttyAMA0`
whilst researching, this: (https://spellfoundry.com/2016/05/29/configuring-gpio-serial-port-raspbian-jessie-including-pi-3-4/) gives a good summary of serial on the Pis

**Comment by john30** [comment]
we've tried to summarize all aspects [in the documentation](https://adapter.ebusd.eu/raspberrypi.en.html)

**Comment by FrigerioSergio** [comment]
Hello.
finally I got the adapter.
Not sure if I am facing the same problems.

I plan to use it on a raspberry pi4 with Home assistant OS and Ebusd 22.4.1 Daemon.
I install the adapter: typing ls -l /dev  I can see TTYAMA0 under Serial1 (and not serial0): 

![image](https://user-images.githubusercontent.com/22483463/210440627-5f60654e-71a1-4f42-9e53-f7a3fa99ab86.png)


I did not have any TTYS0 listed.
Running the Ebusd Daemon the adapter is recognized but I did not see any message.

![image](https://user-images.githubusercontent.com/22483463/210441019-d080eec8-fa35-44b8-afc9-480f93cc795d.png)


 I got an ESERA ebus adapter and with it I can get messages.

Any suggestions? Thanks

Sergio

**Reply by FrigerioSergio** [reply]
Ok. adding inside the config.txt file the sentence: dtoverlay=pi3-miniuart-bt

solve the issue on serial1 Vs serial0 for ttyAMA0

**Reply by FrigerioSergio** [reply]
And solve the issue that I got.
